[rtl,clkmgr] Declare clkmgr at D3

matutem opened the pull request:
The justification is added as a comment in https://github.com/lowRISC/opentitan/issues/22482.

Fixes #22482

matutem commented:
CHANGE AUTHORIZED: hw/ip_templates/clkmgr/data/clkmgr.hjson.tpl

vogelpi reviewed (commented):
Thanks @matutem for doing the signoff work. The PR and the changes to the checklist look good to me. But for documentation purposes we would need to populate the signoff issue and report the findings made during the signoff. Would you be able to do this please?

To get an idea of what is needed, please refer to the `lc_ctrl` D3 signoff issue: https://github.com/lowRISC/opentitan/issues/22480#issuecomment-2254153756

Specifically referring to the lint checklist item: we need to check whether clkmgr was at D3 before and if yes, if the lint waiver file for AscentLint has changed since then. If it hasn't been at D3 before, or if the waiver file changed in the meantime, we need to argue why these waiver rules are required and TC needs to approve this.

matutem commented:
@vogelpi I updated https://github.com/lowRISC/opentitan/issues/22482 with the justification for the signoff. The block was not previously at D3, but I added some explanation for the lint waivers. Let me know if you think it needs more details.

matutem commented:
This also needs another `CHANGE AUTHORIZED` comment to get through.

andreaskurth commented on hw/ip_templates/clkmgr/doc/checklist.md:
```diff
@@ -100,17 +100,17 @@ Security      | [SEC_CM_COUNCIL_REVIEWED][]  | Done        |
-Review        | Signoff date            | Not Started |
+Documentation | [NEW_FEATURES_D3][]     | Done        |
+RTL           | [TODO_COMPLETE][]       | Done        |
+Code Quality  | [LINT_COMPLETE][]       | Done        |
```
```suggestion
Code Quality  | [LINT_COMPLETE][]       | Done        | With waivers approved by TC on 2024-08-08
```
Please stage this change for now. I'll add it to the agenda of the TC meeting and report the result back here.

andreaskurth commented on hw/ip_templates/clkmgr/doc/checklist.md:
```diff
@@ -100,17 +100,17 @@ Security      | [SEC_CM_COUNCIL_REVIEWED][]  | Done        |
+Review        | [REVIEW_DELETED_FF][]   | Done        |
+Review        | [REVIEW_SW_CHANGE][]    | Done        |
+Review        | [REVIEW_SW_ERRATA][]    | Done        |
+Review        | Reviewer(s)             | Done        | @matutem, @vogelpi
```
Please add me (`adk@`) as a reviewer

andreaskurth replied:
Oh, and I think these are OpenTitan handles and should be written `name@` (i.e., `@` at the end) in order to not have GitHub confuse them with GitHub account names

andreaskurth commented:
CHANGE AUTHORIZED: hw/ip_templates/clkmgr/data/clkmgr.hjson.tpl
CHANGE AUTHORIZED: hw/top_earlgrey/ip_autogen/clkmgr/data/clkmgr.hjson

Only `design_stage` changed, which doesn't impact RTL

matutem commented:
@andreaskurth I made the suggested changes. Thanks.

andreaskurth reviewed (approved):
Looks all good to me now. Thanks @matutem!
